programs/mango-v4/src/instructions/token_deposit.rs

```diff
@@ -237,7 +237,7 @@ impl<'a, 'info> DepositCommon<'a, 'info> {
             mango_account: self.account.key(),
             signer: self.token_authority.key(),
             token_index,
-            quantity: amount,
+            quantity: amount_i80f48.to_num::<u64>(),
```

Comment:
This is definitely more correct than the unadjusted `amount`, so approved. Let's discuss out-of-sync issues from fractional deposits separately.